File: activities/06-parsing/Python/Number.py
#Number:top#
#Number:import#

# <numbers>:Number ::= NUM <numbers>
class Number(Numbers): #Number:class#

    className = "Number"
    ruleString = "<numbers>:Number ::= NUM <numbers>"
    numbers = None

    def __init__(numbers):
        #Number:init#
        self.numbers = numbers

#Number#

File: activities/07-parsing/tree/Java/Number.java
//Number:top//
//Number:import//
import java.util.*;

// <tree>:Number ::= NUM
public class Number extends Tree /*Number:class*/ {

    public static final String $className = "Number";
    public static final String $ruleString =
        "<tree>:Number ::= NUM";



    public Number() {
//Number:init//

    }

    public static Number parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<tree>:Number", scn$.lno);
        scn$.match(Token.Match.NUM, trace$);
        return new Number();
    }

//Number//
}
